source/Xunit.Gherkin.Quick/Feature.cs

```diff
@@ -32,39 +30,52 @@ internal void Scenario(string scenarioName)
 
             var parsedScenario = gherkinDocument.Feature.Children.FirstOrDefault(scenario => scenario.Name == scenarioName);
             if (parsedScenario == null)
-                throw new Exception($"Cannot find scenario `{scenarioName}`.");
+                throw new GherkinException($"Cannot find scenario `{scenarioName}`.");
```

Comment:
I don't like exception name honestly. This library is not Gherkin.
I'd rather suggest to use InvalidOperationException or some other concrete type.
Long run, we may see a clear need for another custom exception. For now, I'm not clear how this helps.